source/constructs/api/data_source/crud.py

```diff
@@ -157,7 +158,20 @@ def list_jdbc_instance_source_by_instance_id_account(jdbcConn: JDBCInstanceSourc
                                                           JDBCInstanceSource.account_id == jdbcConn.account_id,
                                                           JDBCInstanceSource.instance_id == jdbcConn.instance_id).all()
 
-def list_jdbc_instance_source(provider_id: int):
+def list_jdbc_instance_source(provider_id: int, condition: QueryCondition):
+    accounts = get_session().query(Account).filter(Account.account_provider_id == provider_id,
+                                                   Account.status == 1).all()
+    account_ids = []
+    for account in accounts:
+        account_ids.append(account.account_id)
+
+    return query_with_condition(get_session().query(JDBCInstanceSource).filter(
+        JDBCInstanceSource.account_id.in_(account_ids),
+        JDBCInstanceSource.account_provider_id == provider_id,
+        or_(JDBCInstanceSource.detection_history_id != -1, JDBCInstanceSource.detection_history_id == None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)